cmd/buildkitd/config.go

```diff
@@ -6,29 +6,32 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/BurntSushi/toml"
 	"github.com/moby/buildkit/cmd/buildkitd/config"
+	"github.com/pelletier/go-toml"
 	"github.com/pkg/errors"
 )
 
-func Load(r io.Reader) (config.Config, *toml.MetaData, error) {
+func Load(r io.Reader) (config.Config, error) {
 	var c config.Config
-	md, err := toml.DecodeReader(r, &c)
+	t, err := toml.LoadReader(r)
 	if err != nil {
-		return c, nil, errors.Wrap(err, "failed to parse config")
+		return c, errors.Wrap(err, "failed to parse config")
 	}
-	return c, &md, nil
+	err = t.Unmarshal(&c)
+	if err != nil {
+		return c, errors.Wrap(err, "failed to parse config")
+	}
+	return c, nil
 }
 
-func LoadFile(fp string) (config.Config, *toml.MetaData, error) {
+func LoadFile(fp string) (config.Config, error) {
 	f, err := os.Open(fp)
```

Comment:
I kept this bit close to the existing implementation, but `go-toml` also has a `toml.LoadFile()` function that we could use. Keeping it close to the existing implementation would keep the errors (roughly) the same, so I decided not to use that.